input_definition.go

```diff
@@ -254,6 +265,56 @@ type InputDefinitionInfo struct {
 	Fields []InputDefinitionField `json:"fields"`
 }
 
+// Validate the InputDefinitionInfo data
+func (i *InputDefinitionInfo) Validate(columnLabel string) error {
+	numPrimaryKey := 0
+	accountRowID := make(map[string]uint64)
+
+	if len(i.Frames) == 0 {
+		return fmt.Errorf("At least one frame is required per Input Definition")
+	}
+
+	for _, frame := range i.Frames {
+		if err := frame.Validate(); err != nil {
+			return err
+		}
+		// TODO frame option validation
+	}
+
+	// Validate columnLabel and duplicate primaryKey.
+	for _, field := range i.Fields {
+		if field.PrimaryKey {
+			numPrimaryKey++
+			if field.Name != columnLabel {
+				return ErrInputDefinitionColumnLabel
+			}
+		}
+		for _, action := range field.Actions {
+			if err := action.Validate(); err != nil {
+				return err
+			}
+			if action.ValueDestination == InputSingleRowBool && action.Frame != "" {
```

Comment:
I think we can remove `&& action.Frame != ""` since action.Validate() return ErrFrameRequired if frame is empty